instant/src/main/java/com/adyen/checkout/instant/InstantPaymentComponent.kt

```diff
@@ -0,0 +1,58 @@
+package com.adyen.checkout.instant
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.adyen.checkout.components.PaymentComponentEvent
+import com.adyen.checkout.components.PaymentComponentProvider
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.base.BasePaymentComponent
+import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+
+/**
+ * Payment component used for handling payment methods that do not require any input from the shopper.
+ *
+ * Component should not be instantiated directly. Instead use the [PROVIDER] object.
+ */
+class InstantPaymentComponent(
+    savedStateHandle: SavedStateHandle,
+    override val delegate: InstantPaymentDelegate,
+    configuration: InstantPaymentConfiguration
+) : BasePaymentComponent<InstantPaymentConfiguration, PaymentComponentState<PaymentMethodDetails>>(
+    savedStateHandle,
+    delegate,
+    configuration
+) {
+
+    override fun getSupportedPaymentMethodTypes() = PAYMENT_METHOD_TYPES
```

Comment:
We could remove this method altogether from the interface since we can rely on the provider now